### PR TITLE
Add CustomLabelEncoder transformer

### LINKED ISSUE #507: Add CustomLabelEncoder transformer
### Problem Description

As a user, it would be useful to have a `LabelEncoder` that uses a custom ordering that I supply.

### Expected behavior

- Add a new transformer called `CustomLabelEncoder`
- This transformer should have the following init parameters:
    - `order`: A list of categories that will appear in the data. The order of this list determines the order of the labels.
    - `add_noise`: Adds uniform noise in the same way as the normal `LabelEncoder`
- Forward transform should work same as normal `LabelEncoder` except use the provided ordering for the labels
- Reverse transform should also work the same

### Validation
- During `fit` if the data contains categories not specified in the list provided in `order` then raise the following error:
    - `Error: Unknown category '{category}'. All possible categories must be defined in the 'order' parameter.`

### PR DESCRIPTION
**Pull request by amontanez24**
resolves #507 

This PR:

- Adds a new `CustomLabelEncoder` class
- Overwrites the class' `fit` method
- Adds unit tests
- Adds integration test
- Skips performance tests, integration tests and quality tests for this transformer since the `order` param cannot be automated for now

### CODE REVIEW

**Comment by amontanez24** [comment]
Still need to add integration tests

File: rdt/transformers/categorical.py [hunk 1]
```diff
@@ -535,3 +535,49 @@ def _reverse_transform(self, data):
...
+
+        Generate a unique integer representation for each category and
+        store them in the `categories_to_values` dict and its reverse
+        `values_to_categories`.
+
+        Args:
+            data (pandas.Series):
+                Data to fit the transformer to.
+        """
+        missing = list(data[~data.isin(self.order)].unique())
```
Comment (fealho) [inline rdt/transformers/categorical.py:572]: This line is converting None to nan, which is breaking things when Nones are involved. Example:

```python3
import pandas as pd
import numpy as np
from rdt.transformers import CustomLabelEncoder

data = pd.DataFrame([None, 1], columns=['column_name'])
transformer = CustomLabelEncoder(order=[None, 1])
transformer.fit(data, 'column_name')
transformed = transformer.transform(data)
```

File: tests/integration/transformers/test_categorical.py [hunk 2]
```diff
@@ -348,3 +348,26 @@ def test_label_encoder_order_by_alphabetical():
     expected = pd.DataFrame([1, 3, 4, 2, 0], columns=['column_name.value'])
     pd.testing.assert_frame_equal(transformed, expected)
     pd.testing.assert_frame_equal(reverse, data)
+
+
+def test_custom_label_encoder():
+    """The the CustomLabelEncoder.
```
Comment (katxiao) [inline tests/integration/transformers/test_categorical.py:354]: Should be "Test the CustomLabelEncoder"

Nit: The comment could be more descriptive

File: tests/unit/transformers/test_categorical.py [hunk 3]
```diff
@@ -1856,3 +1857,70 @@ def test__reverse_transform_add_noise(self):
 
         # Assert
         pd.testing.assert_series_equal(out, pd.Series(['a', 'b', 'c']))
+
+
+class TestCustomLabelEncoder:
+
+    def test___init__(self):
+        """Passed arguments must be stored as attributes."""
```
Comment (katxiao) [inline tests/unit/transformers/test_categorical.py:1865]: nit: Should we follow the convention of starting the description with "Test the ``__init__`` method"?

File: rdt/transformers/categorical.py (hunk 1 above)
Comment (amontanez24) [reply rdt/transformers/categorical.py:572]: I actually can't replicate this. To be safe I can cast all nan types to `np.nan` like in the normal `LabelEncoder`

**Review by fealho — APPROVED** [review]
👍

**Review by katxiao — APPROVED** [review]

**Pull request merged**